### ISSUE TITLE
Simplified StringName literal for single word text

### ISSUE THREAD

**Issue by tokengamedev** [godotengine/godot-proposals#8079]
### Describe the project you are working on

Casual Game in 4.2

### Describe the problem or limitation you are having in your project

Currently the StringName literals are identified as `&"hello godot"`. It is well and fine.
For single word text, can it not be simplified by removing the quotes similar to Node Path like `&hello_godot`

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

It will help in reducing unnecessary quotes as in NodePath 

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Support both the structure for String Name as given below
```gdscript
var str_name := &"Hello"

# Currently the below error outs as 'Expected expression for variable initial value after "=".'
var str_name := &Hello
```

### If this enhancement will not be used often, can it be worked around with a few lines of script?

It will be used mostly as constants and keys to dictionary, where the text is not changing.

### Is there a reason why this should be core and not an add-on in the asset library?

GDscript parser is part of core.

**Comment by dalexeev** [comment]
I had a similar idea, but after some thought I concluded that this syntax might confuse users that there are pointers/references in GDScript:

```gdscript
node_2d.set(&global_scale, Vector2(8, 2.5))
tween.tween_property($Sprite, ^position, Vector2(100, 200), 1)
```

The quotes more clearly indicate that these are just strings:

```gdscript
node_2d.set(&"global_scale", Vector2(8, 2.5))
tween.tween_property($Sprite, ^"position", Vector2(100, 200), 1)
```

**Comment by Mickeon** [comment]
Yeah. It first sounds good on paper but by all practical means, **StringName**s are **String**s and a vast majority of users wouldn't want to even realise there's a distinction between the two, let alone seeing a somewhat "magical" symbol behind what is just one character away from being a property.

**Comment by KoBeWi** [comment]
Duplicate of #4721 ;)

> It will be used mostly as constants and keys to dictionary, where the text is not changing.

Note that currently Dictionaries will implicitly convert StringNames to Strings, making them actually slower: https://github.com/godotengine/godot/issues/68834
There is a pending PR that fixes it, but it's still not merged.

**Comment by tokengamedev** [comment]
I am ok with it being not happening and getting closed, as people coming from C++ background will mistake as a referenced variable rather than StringName (not String), which is actually 
StringName is not a String, rather it is pointer to a string (correct me if I am wrong). 
It is hard to distinguish between two, I will delve into the StringName code sometime, to understand better.

From #4721 I only got, that it was closed because nobody wants to add more colors, although few people liked the proposal. This is disappointing.
